gateway/ln-gateway/src/lib.rs

```diff
@@ -438,81 +495,93 @@ impl Gateway {
+        let (route_hints, _, _) = self.fetch_lightning_route_info().await?;
+
+        let client = self
             .client_builder
-            .create_config(connect, channel_id, node_pub_key, fees)
+            .build(gw_client_cfg.clone(), self.lnrpc.clone(), task_group)
             .await?;
 
-        let client = Arc::new(
-            self.client_builder
-                .build(
-                    gw_client_cfg.clone(),
-                    self.decoders.clone(),
-                    self.module_gens.clone(),
-                )
-                .await
-                .expect("Failed to build gateway client"),
+        let (gateway, _) = client.get_first_module::<GatewayClientModule>(&KIND);
+
+        let registration = gateway.to_gateway_registration_info(
+            route_hints.clone(),
+            GW_ANNOUNCEMENT_TTL,
+            self.api.clone(),
         );
 
-        let actor = self
-            .load_actor(client.clone(), route_hints, channel_id)
-            .await
-            .map_err(|e| {
-                GatewayError::Other(anyhow::anyhow!("Failed to connect federation {}", e))
-            })?;
+        self.register_client(client, federation_id, channel_id, route_hints)
+            .await?;
 
         let dbtx = self.gatewayd_db.begin_transaction().await;
-        if let Err(e) = self
-            .client_builder
-            .save_config(client.config(), payload.connect, dbtx)
-            .await
-        {
-            warn!(
-                "Failed to save default federation client configuration: {}",
-                e
-            );
-        }
-
-        let federation_info = actor.get_info()?;
+        self.client_builder
+            .save_config(gw_client_cfg.clone(), dbtx)
+            .await?;
 
-        Ok(federation_info)
+        Ok(FederationInfo {
+            federation_id,
+            registration,
+        })
     }
 
     async fn handle_get_info(&self, _payload: InfoPayload) -> Result<GatewayInfo> {
-        let actors = self.actors.read().await;
-        let mut federations: Vec<FederationInfo> = Vec::new();
-        for actor in actors.values() {
-            federations.push(actor.get_info()?);
-        }
+        let mut federations = Vec::new();
+        let federation_clients = self.clients.read().await.clone().into_iter();
+        let (route_hints, node_pub_key, alias) = self.fetch_lightning_route_info().await?;
+        for (federation_id, client) in federation_clients {
+            // TODO: We're reconstructing these registrations, which could have changed in
+            // the meantime, which might break some tests if they're expecting
+            // the same values as the previous registration
+            let (gateway, _) = client.get_first_module::<GatewayClientModule>(&KIND);
+            let registration = gateway.to_gateway_registration_info(
+                route_hints.clone(),
+                GW_ANNOUNCEMENT_TTL,
+                self.api.clone(),
```

Comment:
The `valid_until` value here won't be the same as what the registration with the federation since internally this calls `fedimint_core::time::now()`

Reply:
Yes this is a bug. Ultimately I don't think `info` should return the registrations. The only source of truth of that is the federation, so we could just query the federation here, but ideally I think this function shouldn't be dependent on the federation. So I think this should be revisited and the amount of data returned to the client should be slimmed down.

Reply:
https://github.com/fedimint/fedimint/issues/2651